### DISCUSSION TITLE
Connection Details from Spring AMQP?

### DISCUSSION THREAD

**Discussion by lost-carrier** [rabbitmq/hop#240] [General]
Maybe I'm just missing a bit of documentation here, but - since this is recommended by Spring AMQP project:

What is the most elegant way to ome from a Spring AMQP wired connection (-> `RabbitTemplate`) to a HOP one?

The Spring AMQP `ConnectionFactory` or `Connection` would offer host, post, virtualHost, username, but NO password. The thing is that I might not actually know where Spring is getting the connection details from (imagine Docker or CloudFoundry context). There is not by chance anything like `Client hopClient = new Client(rabbitTemplate.getConnectionFactory());`? (...at least I could not find anything useful...)

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This is a question for Spring AMQP maintainers. They seem to [recommend using Hop directly](https://docs.spring.io/spring-amqp/docs/current/reference/html/#management-rest-api) but then have `RabbitAdmin`. I don't see anything relevant [in these tests](https://github.com/spring-projects/spring-amqp/blob/2cded99e338899774c64a0d9f269df861f4383a1/spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitAdminTests.java).

**Reply by lost-carrier** [reply]
`RabbitAdmin` is very limited. I'm particularly interested in `getBindings()` which appears not to be available on `RabbitAdmin`, but found it on `RabbitManagementTemplate`, which in turn has been `@Deprecated` "since 2.1 in favor of direct `Client` usage."

https://docs.spring.io/spring-amqp/docs/2.1.8.RELEASE/api/index.html?org/springframework/amqp/rabbit/core/RabbitManagementTemplate.html

**Reply by michaelklishin** [reply]
It's a reasonable feature request for `RabbitAdmin` or some other interface in Spring AMQP to expose the HTTP API client it ultimately relies on. The only possible change to Hop that might entail is a new constructor of some kind but since Spring folks never asked for one to date, it should not be necessary in theory.
